### ISSUE TITLE
missing ini files on new builds

### ISSUE THREAD

**Issue by granitepenguin** [wwivbbs/wwiv#968]
init --initialize is setting up wwiv.ini, it should also set up wwivd.ini and net.ini with some defaults

wwivd.ini
```ini
[WWIVD]
telnet_port = 23
ssh_port = 2222
binkp_port = 24554
end_node = 4
```

net.ini
```ini
[networkb]
skip_net = N
crc = Y
```

**Comment by wwiv** [comment]
actually @granitepenguin , it just renames wwivini.txt to wwiv.ini.

Should we bundle defaults for these?

**Comment by wwiv** [comment]
Also I'd rather init not do this rename, but the install script does.  Also we could have an upgrade script too...

**Comment by granitepenguin** [comment]
I would think so.  What was the reasoning behind the rename in the first place?

**Comment by granitepenguin** [comment]
I don't have a problem having the install.sh doing it instead of init, either.

**Comment by wwiv** [comment]
so if you unzip your wwiv*.zip you don't stomp your ini

**Comment by wwiv** [comment]
let's move the rename to the install.{sh,cmd}

**Comment by granitepenguin** [comment]
per discussion, going to create an inifiles directory in bbs/admin and set up the build to use the inifiles.zip

**Comment by granitepenguin** [comment]
I'm most of the way there, but I can't figure out why this is happening:

017-01-02 20:48:11,412 ERROR TEMP_DIRECTORY is not set! Unable to create CONFIG.OVR

It's coming from CreateConfigOvr() in init.cpp, but it's in this part of the code:

```c
  bool need_to_initialize = !File::Exists(CONFIG_DAT) || forced_initialize;

  out->Cls(ACS_CKBOARD);
  out->window()->SetColor(SchemeId::NORMAL);

  if (need_to_initialize) {
    out->window()->Bkgd(' ');
    if (!new_init(out->window(), bbsdir, !forced_initialize)) {
      return 2;
    }
  }

  Config config(bbsdir);
  upgrade_datafiles_if_needed(config);
  CreateConfigOvr(bbsdir);
```

The reason why I'm confused is I have created an inifiles.zip for new_init and I've updated newinit.cpp to extract it.  

```c
  window->Printf(".\n");

  window->SetColor(SchemeId::PROMPT);
  window->Puts("Copying String and Miscellaneous files.\n");
  window->SetColor(SchemeId::NORMAL);

  window->Printf(".\n");


  window->SetColor(SchemeId::PROMPT);
  window->Puts("Decompressing archives.  Please wait");
  window->SetColor(SchemeId::NORMAL);

  if (unzip_files) {
    unzip_file("inifiles.zip", bbsdir);
    unzip_file("gfiles.zip", "gfiles");
    unzip_file("scripts.zip", "scripts");
    unzip_file("data.zip", "data");
    unzip_file("regions.zip", "data");
    unzip_file("zip-city.zip", "data");
  }
```

as far as I can tell, the zip files should be extracted before CreateConfigOvr(bbsdir) is called.  My guess is I am not getting the directory right and unzip_file is not working correctly.

I tried ``unzip_file("inifiles.zip", ".");`` first, which didn't work, so then I tried ``unzip_file("inifiles.zip", bbsdir);`` since bbsdir is a std::string, but maybe I'm not getting the right kind of string?

**Comment by granitepenguin** [comment]
hmm... apparently ``if (unzip_files)`` is false; that's why it isn't working.  It's skipping all the zip files, not just the inifiles.zip.  Just need to figure out why that's the case.

**Comment by granitepenguin** [comment]
I think this is wrong in init.cpp:

```c
  if (need_to_initialize) {
    out->window()->Bkgd(' ');
    if (!new_init(out->window(), bbsdir, !forced_initialize)) {
      return 2;
    }
  }
```

forced_initialize is negated, which means the corresponding unzip_files variable is false and the zip files are never unzipped.  that isn't what we want, because if need_to_initialize is true, it's because either we are missing config.dat, or forced_initialize is true.  

I see that this was intentional in ca854ee1b9ad178a6ff492580e6cfc7ae9825507 so the installer can do it instead, but without an installer on the Windows side, we are stuck.  I think we should re-enable this in init for the time being.

**Comment by wwiv** [comment]
sgtm

**Comment by granitepenguin** [comment]
Fixed in 0cce9f7913be2d36ffef45ac14c7b92b36056166
